pkg/jobrunaggregator/jobruntestcaseanalyzer/analyzer.go

```diff
@@ -0,0 +1,404 @@
+package jobruntestcaseanalyzer
+
+import (
+	"context"
+	"encoding/xml"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"gopkg.in/yaml.v2"
+
+	"k8s.io/apimachinery/pkg/util/clock"
+
+	"github.com/openshift/ci-tools/pkg/jobrunaggregator/jobrunaggregatorapi"
+	"github.com/openshift/ci-tools/pkg/jobrunaggregator/jobrunaggregatorlib"
+	"github.com/openshift/ci-tools/pkg/junit"
+)
+
+type testIdentifier struct {
+	testSuites []string
+	testName   string
+}
+
+var (
+	installTestSuites     = []string{"cluster install"}
+	installTest           = "install should succeed: overall"
+	installTestIdentifier = testIdentifier{installTestSuites, installTest}
+)
+
+// TestCaseChecker checks if a test passes certain criteria across all job runs
+type TestCaseChecker interface {
+	// CheckTestCase returns a test case based on whether a test has passed certain criteria across job runs
+	CheckTestCase(ctx context.Context, jobRuns []jobrunaggregatorapi.JobRunInfo) *junit.TestCase
+}
+
+type minimumRequiredPassesTestCaseChecker struct {
+	id                     testIdentifier
+	testNameSuffix         string
```

Comment:
what is this and what is it for?

Reply:
Need the identifiers (platform, network etc) for the test names inside TestCaseChecker. I can add another function (SetTestNameSuffix()) to the interface if it is better. Or another better way in go?

Reply:
Please add godoc explaining this.